x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/hooks/use_esql_index.ts

```diff
@@ -30,7 +30,7 @@ export const useEsqlIndex = (
       return '';
     }
     const esqlQuery = typeof query === 'string' && isEsqlRule(ruleType) ? query : undefined;
-    return getIndexPatternFromESQLQuery(esqlQuery);
+    return getIndexPatternFromESQLQueryDeprecated(esqlQuery);
```

Comment:
ℹ️ Here we need to use the async getIndexPatternFromESQLQuery instead but I prefer the code owners to handle it